pkg/bbgo/activeorderbook.go

```diff
@@ -277,25 +285,48 @@ func (b *ActiveOrderBook) Print() {
 func (b *ActiveOrderBook) Update(orders ...types.Order) {
 	hasSymbol := len(b.Symbol) > 0
 	for _, order := range orders {
-		if hasSymbol && b.Symbol == order.Symbol {
-			b.orders.Update(order)
+		if hasSymbol && b.Symbol != order.Symbol {
+			continue
 		}
+
+		b.orders.Update(order)
 	}
 }
 
 func (b *ActiveOrderBook) Add(orders ...types.Order) {
 	hasSymbol := len(b.Symbol) > 0
 	for _, order := range orders {
-		if hasSymbol && b.Symbol == order.Symbol {
+		if hasSymbol && b.Symbol != order.Symbol {
+			continue
+		}
+
+		b.add(order)
+	}
+}
+
+// add the order to the active order book and check the pending order
+func (b *ActiveOrderBook) add(order types.Order) {
+	if pendingOrder, ok := b.pendingOrderUpdates.Get(order.OrderID); ok {
+		if pendingOrder.UpdateTime.Time().After(order.UpdateTime.Time()) {
```

Comment:
There's a rare condition that a pending order and a new coming order are updated at the same timestamp (in millisecond or even in second). 
Another way is to compare the filled amount or the state change.